src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -746,14 +746,46 @@ def resample(self, timestamp: datetime) -> Sample[Quantity]:
         # resort to some C (or similar) implementation.
         relevant_samples = list(itertools.islice(self._buffer, min_index, max_index))
         if not relevant_samples:
-            _logger.warning("No relevant samples found for: %s", self._name)
+            self._log_no_relevant_samples(minimum_relevant_timestamp, timestamp)
```

Comment:
We are probably always going to enable warnings, but just in case and since that function seems to be non-trivial, maybe you want to [check if the log level is enabled](https://docs.python.org/3/library/logging.html#logging.Logger.isEnabledFor) so you don't call it at all if it's not:

```py
        if not relevant_samples and _logger.isEnabledFor(logging.WARNING):
```

Or maybe it makes more sense to add the check inside the function, to have the levels in the same place (at the expense of a function call that will just return).